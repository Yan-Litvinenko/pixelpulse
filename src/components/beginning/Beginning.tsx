import React from 'react';
import beginnintAvif from '../../assets/images/beginning.avif';
import beginnintWebp from '../../assets/images/beginning.webp';
import beginningJpg from '../../assets/images/beginning.jpg';
import styles from './Beginning.module.scss';
import { Frame } from '../frame/Frame';

const Beginning = (): React.JSX.Element => {
    return (
        <section className={styles.beginning} data-testid="beginning">
            <div className={styles.beginning__inner}>
                <>
                    <div className={styles.beginning__box_title}>
                        <h2 className={styles.beginning__title}>
                            swimming through a vast network of interconnected
                            devices and servers, spreading joy and whimsy to
                            users across the globe
                        </h2>
                        <h3 className={styles.beginning__subtitle}>
                            Artwork generated with midjourney
                        </h3>
                    </div>

                    <picture className={styles.beginning__background_box}>
                        <source srcSet={beginnintAvif} />
                        <source srcSet={beginnintWebp} />
                        <img
                            src={beginningJpg}
                            alt="Artwork of a whale swimming through a network of interconnected devices"
                            className={styles.beginning__background}
                        />
                        <Frame className={styles.beginning__background_frame} />
                    </picture>
                </>
            </div>
        </section>
    );
};

export { Beginning };
