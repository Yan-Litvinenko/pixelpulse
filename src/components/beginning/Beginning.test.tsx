import React from 'react';
import { Beginning } from './Beginning';
import { render, screen } from '@testing-library/react';

describe('Beginning component', (): void => {
    test('Render', (): void => {
        const component = render(<Beginning />);

        expect(screen.getByTestId('beginning')).toBeInTheDocument();
        expect(component).toMatchSnapshot();
    });
});
