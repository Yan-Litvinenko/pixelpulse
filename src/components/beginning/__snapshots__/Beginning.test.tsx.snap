// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beginning component Render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        data-testid="beginning"
      >
        <div>
          <div>
            <h2>
              swimming through a vast network of interconnected devices and servers, spreading joy and whimsy to users across the globe
            </h2>
            <h3>
              Artwork generated with midjourney
            </h3>
          </div>
          <picture>
            <source
              srcset="test-file-stub"
            />
            <source
              srcset="test-file-stub"
            />
            <img
              alt="Artwork of a whale swimming through a network of interconnected devices"
              src="test-file-stub"
            />
            <div>
              <svg
                fill="none"
                height="0.5625rem"
                viewBox="0 0 9 9"
                width="0.5625rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 9L1 1L9 1"
                  stroke="current-color"
                  stroke-width="2"
                />
              </svg>
              <svg
                fill="none"
                height="0.5625rem"
                viewBox="0 0 9 9"
                width="0.5625rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 1L8 1L8 9"
                  stroke="current-color"
                  stroke-width="2"
                />
              </svg>
              <svg
                fill="none"
                height="0.5625rem"
                viewBox="0 0 9 9"
                width="0.5625rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 0L8 8L1.04907e-06 8"
                  stroke="current-color"
                  stroke-width="2"
                />
              </svg>
              <svg
                fill="none"
                height="0.5625rem"
                viewBox="0 0 9 9"
                width="0.5625rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 8L1 8L0.999999 6.99382e-07"
                  stroke="current-color"
                  stroke-width="2"
                />
              </svg>
            </div>
          </picture>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      data-testid="beginning"
    >
      <div>
        <div>
          <h2>
            swimming through a vast network of interconnected devices and servers, spreading joy and whimsy to users across the globe
          </h2>
          <h3>
            Artwork generated with midjourney
          </h3>
        </div>
        <picture>
          <source
            srcset="test-file-stub"
          />
          <source
            srcset="test-file-stub"
          />
          <img
            alt="Artwork of a whale swimming through a network of interconnected devices"
            src="test-file-stub"
          />
          <div>
            <svg
              fill="none"
              height="0.5625rem"
              viewBox="0 0 9 9"
              width="0.5625rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 9L1 1L9 1"
                stroke="current-color"
                stroke-width="2"
              />
            </svg>
            <svg
              fill="none"
              height="0.5625rem"
              viewBox="0 0 9 9"
              width="0.5625rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 1L8 1L8 9"
                stroke="current-color"
                stroke-width="2"
              />
            </svg>
            <svg
              fill="none"
              height="0.5625rem"
              viewBox="0 0 9 9"
              width="0.5625rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 0L8 8L1.04907e-06 8"
                stroke="current-color"
                stroke-width="2"
              />
            </svg>
            <svg
              fill="none"
              height="0.5625rem"
              viewBox="0 0 9 9"
              width="0.5625rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 8L1 8L0.999999 6.99382e-07"
                stroke="current-color"
                stroke-width="2"
              />
            </svg>
          </div>
        </picture>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
